Sources/DistributedActors/Pattern/WorkerPool.swift

```diff
@@ -44,19 +54,79 @@ public struct WorkerPool<Message: ActorMessage> {
         /// The worker pool will terminate itself if all of its static workers have terminated.
         /// You may death-watch the worker pool in order to react to this situation, e.g. by spawning a replacement pool,
         /// or gracefully shutting down your application.
-        case `static`([_ActorRef<Message>])
+        case `static`([Worker])
     }
 
     // ==== ------------------------------------------------------------------------------------------------------------
     // MARK: State shared across all states of the WorkerPool
 
-    var settings: WorkerPoolSettings<Message>
+    var settings: WorkerPoolSettings<Worker>
     var selector: Selector {
         self.settings.selector
     }
-
-    init(settings: WorkerPoolSettings<Message>) {
+    
+    /// The worker pool. Worker will be selected round-robin.
+    private var workers: [Weak<Worker>] = []
+    private var roundRobinPos = 0
+    
+    /// Boolean flag to help determine if pool becomes empty because at least one worker has terminated.
+    private var hasTerminatedWorkers = false
+
+    init(settings: WorkerPoolSettings<Worker>, system: ActorSystem) async {
         self.settings = settings
+        self.actorSystem = system
+
+        switch self.selector {
+        case .dynamic(let key):
+            Task {
+                for await worker in await self.actorSystem.receptionist.subscribe(to: key) {
+                    system.log.log(level: self.settings.logLevel, "Got listing member for \(self.selector): \(worker)")
+                    self.workers.append(Weak(worker))
+                    watchTermination(of: worker) { self.onWorkerTerminated(id: $0) }
+                }
+            }
```

Comment:
need to store the task in the actor, and as we `deinit` cancel it (`task.cancel()`)

Reply:
Amended https://github.com/apple/swift-distributed-actors/pull/912/commits/0ed16f138acdfabc6e7891250a385e11b579c4bf